### PR TITLE
Fix log collection with new log path locations

### LINKED ISSUE #3100: [ceph] regression: cannot collect log files for package based deployments
#3092 brought some changes with regards to log paths based on Ceph versions. However, it made logs collected in the previous revisions disappear from sosreport.

On a Ceph OSD node with Ubuntu for example,
```
$ ceph --version
ceph version 17.2.0 (43e2e60a7559d3f46c9d53f1ca875fd499a1e35e) quincy (stable)

$ ./bin/sos report --batch -v --label "$(git rev-parse --short HEAD)"
```

With 699654ad,
```
$ grep /var/log/ceph sosreport-$HOSTNAME-699654ad-$(date -Idate)-*/sos_logs/sos.log
2023-01-04 01:15:39,361 INFO: [plugin:ceph_osd] collecting path '/var/log/ceph/ceph-volume.log'
2023-01-04 01:15:39,361 DEBUG: [plugin:ceph_osd] copying path '/var/log/ceph/ceph-volume.log' to archive:'/var/log/ceph/ceph-volume.log'
2023-01-04 01:15:39,362 INFO: [plugin:ceph_osd] collecting path '/var/log/ceph/ceph-osd.1.log'
2023-01-04 01:15:39,362 DEBUG: [plugin:ceph_osd] copying path '/var/log/ceph/ceph-osd.1.log' to archive:'/var/log/ceph/ceph-osd.1.log'
```
-> OSD logs are collected.
[sosreport-epic-adder-699654ad-2023-01-04-wnglymy.tar.gz](https://github.com/sosreport/sos/files/10340864/sosreport-epic-adder-699654ad-2023-01-04-wnglymy.tar.gz)

With 9c60afed,
```
$ grep /var/log/ceph sosreport-$HOSTNAME-9c60afed-$(date -Idate)-*/sos_logs/sos.log
[empty]
```
-> no OSD logs are collected.
[sosreport-epic-adder-9c60afed-2023-01-04-djnwwav.tar.gz](https://github.com/sosreport/sos/files/10340865/sosreport-epic-adder-9c60afed-2023-01-04-djnwwav.tar.gz)

My gut feeling is that the condition in the following part is not strictly tied to Ceph versions actually but rather tied to deployment methods such as package based, ceph-ansible container based, or cephadm container based as I commented in https://github.com/sosreport/sos/pull/3099#discussion_r1060417179

https://github.com/sosreport/sos/blob/9c60afed25c6cd8ecbd25c52a403376262a121d2/sos/report/plugins/ceph_osd.py#L44-L69

### PR DESCRIPTION
**Pull request by swalladge**
WIP

Fixes #3100 

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [ ] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [ ] Is the subject and message clear and concise?
- [ ] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [ ] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by nkshirsagar** [comment]
LGTM

**Review by TurboTurtle — COMMENTED** [review]
This all is looking good to me thus far. As for the version split for ceph ids in `ceph_mon` - those are just the ceph versions I am somewhat familiar with. If ceph 15 is still used widely and can fit in either of those approaches, feel free to make the appropriate change there as well.

**Comment by nkshirsagar** [comment]
> This all is looking good to me thus far. As for the version split for ceph ids in `ceph_mon` - those are just the ceph versions I am somewhat familiar with. If ceph 15 is still used widely and can fit in either of those approaches, feel free to make the appropriate change there as well.

~~~
root@juju-bc74fc-cephoctopus-0:/home/ubuntu# ceph -v
ceph version 15.2.17 (8a82819d84cf884bd39c17e3236e0632ac146dc4) octopus (stable)
root@juju-bc74fc-cephoctopus-0:/home/ubuntu# ps -u ceph -o args
COMMAND
/usr/bin/ceph-mon -f --cluster ceph --id juju-bc74fc-cephoctopus-0 --setuser ceph --setgroup ceph
/usr/bin/ceph-mgr -f --cluster ceph --id juju-bc74fc-cephoctopus-0 --setuser ceph --setgroup ceph


root@juju-bc74fc-cephoctopus-0:/home/ubuntu/sos/sos# git diff
diff --git a/sos/report/plugins/ceph_mon.py b/sos/report/plugins/ceph_mon.py
index 81d3cb12..f281dc6d 100644
--- a/sos/report/plugins/ceph_mon.py
+++ b/sos/report/plugins/ceph_mon.py
@@ -164,7 +164,7 @@ class CephMON(Plugin, RedHatPlugin, UbuntuPlugin):
     def get_ceph_ids(self):
         ceph_ids = []
         # ceph version 14 correlates to RHCS 4
-        if self.ceph_version == 14:
+        if self.ceph_version == 14 or self.ceph_version == 15:
             # Get the ceph user processes
             out = self.exec_cmd('ps -u ceph -o args')


root@juju-bc74fc-cephoctopus-0:/tmp/sosreport-juju-bc74fc-cephoctopus-0-2023-01-11-lbltfgx/sos_commands/ceph_mon/json_output# ls *mon_status*
ceph_tell_mon.juju-bc74fc-cephoctopus-0_mon_status

The ceph_tell_mon.juju-bc74fc-cephoctopus-0_mon_status file is not collected on octopus without the above patch.
~~~

File: sos/report/plugins/ceph_mon.py [hunk 1]
```diff
@@ -161,7 +151,7 @@ def get_ceph_version(self):
         )
         return 0
 
-    def get_ceph_ids(self):
+    def get_ceph_ids(self):  # To check
```
Comment (TurboTurtle) [inline sos/report/plugins/ceph_mon.py:154]: Is this the last part to review before moving this PR out of draft?

Anything we can do to help with this, so we can make sure these changes are included in sos-4.5 which closes in about 1.5 weeks?

Comment (TurboTurtle) [reply sos/report/plugins/ceph_mon.py:154]: Bumping this again. We'd like to get this into sos-4.5. Is there anything left to check here, @swalladge?

Comment (nobuto-m) [reply sos/report/plugins/ceph_mon.py:154]: Hi @TurboTurtle, my apologies I was distracted by other stuff and wasn't able to follow this up. We will double check what we need for the short term to get it into 4.5 and will update here.

**Comment by TurboTurtle** [comment]
#3122 merged as successor to this draft PR. Closing this one. Thanks again for the investigation on this.

**Pull request closed without merging**